app/hooks/throttled.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {throttle} from 'lodash';
+import {useMemo, useRef} from 'react';
+
+const useThrottled = <T extends any[]>(callback: (...args: T) => void, time: number) => {
```

Comment:
Not sure about the types, so open to discussion.

In a similar hook (`usePreventDoubletap`) I used this typing:
```
 <T extends Function>(callback: T)
```
And then this for the body of the callback created in the body:
```
(...args: unknown[]) => 
```
Not sure what is the best approach right now, so happy to discuss.

Reply:
Really liked this approach since it makes sense.

The args of course are unknown, thanks for the idea.

Just updated the code, can you take a look?